packages/kit/src/exports/vite/index.js

```diff
@@ -218,6 +218,20 @@ function kit({ svelte_config }) {
 
 			const generated = path.posix.join(kit.outDir, 'generated');
 
+			// This ensures that esm-env is inlined into the server output with the
+			// export conditions resolved correctly through Vite. This prevents adapters
+			// that bundle later on from resolving the export conditions incorrectly
+			// and for example include browser-only code in the server output
+			// because they for example use esbuild.build with `platform: 'browser'`
+			const noExternal = ['esm-env'];
+
+			// Vitest bypasses Vite when loading external modules, so when it is
+			// detected we take a tiny performance hit to preserve correctness.
+			// See https://github.com/sveltejs/kit/pull/9172
```

Comment:
I'd probably update "correctness" to describe exactly what breaks
```suggestion
			// Vitest bypasses Vite when loading external modules, so we bundle
			// when it is detected to keep our virtual modules working.
			// See https://github.com/sveltejs/kit/pull/9172
			// and https://vitest.dev/config/#deps-registernodeloader
```